Rocket.Chat/Controllers/Chat/MessagesViewController.swift

```diff
@@ -68,14 +87,11 @@ extension MessagesViewController {
         if let size = viewSizingModel.size(for: sectionViewModel.differenceIdentifier) {
             return size
         } else {
-            let sizingCell = BasicMessageCell.sizingCell
-            sizingCell.prepareForReuse()
-            sizingCell.viewModel = sectionViewModel
-            sizingCell.configure()
-            sizingCell.setNeedsLayout()
-            sizingCell.layoutIfNeeded()
-
-            let size = sizingCell.contentView.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize)
+            guard let sizingCell = UINib(nibName: sectionViewModel.relatedReuseIdentifier, bundle: nil).instantiate() as? SizingCell else {
```

Comment:
SizingCell protocol was created in order to avoid having to check the specific object of the section to manually reference the static sizing cell of each type. 

The only con is: for this to work we *must* assign the cell's nib name to its `reuseIdentifier`.  Which is quite easy since we can use `String(describing:_)` method on the cell class, and instead of returning `.zero` we can throw a fatal error, that way we will make sure this will only happen at development time and we can set a message enforcing that the `reuseIdentifier` must be the same as the nib name. 


What do you guys think about this solution?

Reply:
I think that makes perfect sense for now, thanks for letting us know. 👍